### PR TITLE
SimDataFormats/GeneratorProducts : Changes needed to compile with macOS clang and libc++

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by davidlange6** [comment]
this looks like something that needs a change at the tool file level. we should not have ldflags options in build files. @smuzaffar

**Comment by smuzaffar** [comment]
@gartung , did you modify the gcc-toolfile for osx build? or you build gcc from externals?
Its better to have separate spec/toolfile for fortran on mac which should set/add its lib64 in LIBDIR.

**Comment by gartung** [comment]
Spack defines the compilers. I used the spack wrappers $CC $CXX and $F77 in gcc-toolfile. Actually I did not set them and picked them up from the spack environment.

**Comment by gartung** [comment]
https://github.com/gartung/scram-tool-templ/blob/master/selected/gcc-f77compiler.xml

**Comment by gartung** [comment]
So I could add something like <use name="gfortran"> and create a tool config file for gfortan with the -L flag?

**Comment by gartung** [comment]
and add \<use name="gfortran"\> to hepmc.xml

**Comment by smuzaffar** [comment]
I would suggest to have clang-f77compiler.xml tool file and then scram will automatically add it when it will find .F file in the package.

**Comment by smuzaffar** [comment]
ah ok. so you were not use special clang-*compiler.xml file. So scram should have then added dependency on your gcc-f77compiler. If that was not working then I should fix scram.

**Comment by gartung** [comment]
Scram correctly add the -lgfortran option but there is no -L option added probably because spack is providing the wrapper script.

**Comment by smuzaffar** [comment]
did you try adding
```
<flags LDFLAGS="-L$(GCC_F77COMPILER_BASE)/lib64"/>
OR
<environment name="LIBDIR" default="$GCC_F77COMPILER_BASE/lib64"/>
```
in the gcc-f77compiler.xml?

After adding any of these I do see that scram does add it for linking.

**Comment by gartung** [comment]
I have \<flags LDFLAGS="-L$(GCC_F77COMPILER_BASE)/lib64"/\>
in gcc-f77compiler.xml now. I am trying the build with this and the -L flags removed from the package Buildfile.xml

File: SimDataFormats/GeneratorProducts/src/classes_def.xml [hunk 1]
```diff
@@ -16,7 +16,8 @@
     <class name="HepMC::GenCrossSection" ClassVersion="10">
      <version ClassVersion="10" checksum="920043842"/>
     </class>
-	<class name="HepMC::WeightContainer" ClassVersion="14">
+	<class name="HepMC::WeightContainer" ClassVersion="15">
+  <version ClassVersion="15" checksum="376377869"/>
```
Comment (smuzaffar) [inline SimDataFormats/GeneratorProducts/src/classes_def.xml]: this will only work with newer HepMC version .... correct?

**Comment by gartung** [comment]
I think so. I can test without the change again.

**Comment by gartung** [comment]
Remaining change can wait until libhepmc is updated in main CMSSW build.

**Comment by gartung** [comment]
The checksum changes could be because of the two commented out lines.

File: SimDataFormats/GeneratorProducts/src/classes_def.xml (hunk 1 above)
Comment (gartung) [inline SimDataFormats/GeneratorProducts/src/classes_def.xml]: Checksum change is from removal of lines below. They cause warning which fails genreflex generation:
Warning: Unused class rule: std::iterator<std::forward_iterator_tag,HepMC::GenParticle*>
Warning: Unused class rule: std::iterator<std::forward_iterator_tag,HepMC::GenVertex*>

Comment (gartung) [inline SimDataFormats/GeneratorProducts/src/classes_def.xml]: Checksum is because of newer HepMC or root version.

**Comment by gartung** [comment]
Checksum change  is because of newer HepMC or root version.

**Pull request closed without merging**